### PR TITLE
fix spurious 'item not found' error in `std help`

### LINKED ISSUE #9154: `std help` -- extra "help::item_not_found" error
### Describe the bug

I get valid output from `std help, but then an item not found error.  How rude!

### How to reproduce

```
〉std help table
Render the table.

If the table contains a column called 'index', this column is used as the table index instead of the usual continuous index.

---------------   clip --------------------

  Render data in table view (collapsed)
  > [[a b]; [1 2] [2 [4 4]]] | table --collapse
  ╭───┬───┬───╮
  │ # │ a │ b │
  ├───┼───┼───┤
  │ 0 │ 1 │ 2 │
  │ 1 │ 3 │ 4 │
  ╰───┴───┴───╯

Error: 
  × std::help::item_not_found
     ╭─[help:709:1]
 709 │ 
 710 │     let item = ($item | str join " ")
     ·     ─┬─
     ·      ╰── item not found
 711 │ 
     ╰────

```

### Expected behavior

std help succeeded in displaying some help, no reason to issue an error message.

### Screenshots

_No response_

### Configuration

| key                | value                                    |
| ------------------ | ---------------------------------------- |
| version            | 0.79.1                                   |
| branch             | param_defaults_8928_v2                   |
| commit_hash        | 73981dffcc529621cbf809e9e15a0072b6e9ad92 |
| build_os           | linux-x86_64                             |
| build_target       | x86_64-unknown-linux-gnu                 |
| rust_version       | rustc 1.67.1 (d5a82bbd2 2023-02-07)      |
| rust_channel       | 1.67.1-x86_64-unknown-linux-gnu          |
| cargo_version      | cargo 1.67.1 (8ecd4f20a 2023-01-10)      |
| build_time         | 2023-05-09 10:49:10 -04:00               |
| build_rust_channel | debug                                    |
| features           | default, zip                             |
| installed_plugins  |                                          |

### Additional context

This is a regression in recent builds -- I'm not pointing the finger, but the code at the bottom of `help.nu` is depending on `show_command()` to return a value, but the latter doesn't set a return value on all success paths.

### PR DESCRIPTION
**Pull request by bobhy**
…ses).

# Description

See title.  
Fixes #9154 
(despite name of branch)

# User-Facing Changes

n/a
# Tests + Formatting

# After Submitting

### CODE REVIEW

**Review by amtoine — CHANGES_REQUESTED** [review]
noice, good catch :+1: 

> **Note**
> i think this will disappear in the end when replacing the `print` by a full help page construction, but for now it's nice to get rid of these ugly errors :relieved: 

just need to apply the following change
```diff
diff --git a/crates/nu-std/lib/help.nu b/crates/nu-std/lib/help.nu
index 1a2d325cd..ca074d96b 100644
--- a/crates/nu-std/lib/help.nu
+++ b/crates/nu-std/lib/help.nu
@@ -147,7 +147,7 @@ def show-module [module: record] {
         print $"This module (ansi cyan)does not export(ansi reset) environment."
     } else {
         print $"This module (ansi cyan)exports(ansi reset) environment."
-        view source $module.env_block
+        print (view source $module.env_block)
     }
 }
```
otherwise, `std help modules` is not equivalent with its `main` counterpart :scream:

**Comment by bobhy** [comment]
Umm, what bug does this recommended change fix?  I don't think it's related to the problem of the spurious error message?

**Comment by fdncred** [comment]
what do these `" "`'s fix the problem? Seems like we're patching a bug with a hack (even though it works).

**Comment by amtoine** [comment]
`std help std` is not the same on latest `main` and on the tip of this branch => the
```
{
    use dirs *
}
``` block
- does appear on `main`
- does not appear on your branch anymore

the patch above should fix this :+1:

**Comment by amtoine** [comment]
@fdncred, i think the issue is that the `show-*` commands do not return anything and rather only `print` stuff :thinking: 

i plan to make them return a full help page to print it later, that should fix this bug in the end :+1: 
but for now, that's a quick fix, for sure :confused:

**Comment by bobhy** [comment]
Took a bit to see what problem your patch was fixing!  And I can't figure out how that change didn't get merged into my branch automatically.  But I think I've got it now:
```
〉std help std
Help pages from external command std:
No manual entry for std
std.nu, `used` to load all standard library components

Module: std

Exported commands:
    dirs add (std dirs add), dirs next (std dirs next), dirs prev (std dirs prev), dirs drop (std dirs drop), dirs show (std dirs show), dirs goto (std dirs goto), help modules (std help modules), help aliases (std help aliases), help externs (std help externs), help operators (std help operators), help commands (std help commands), help (std help), iter find (std iter find), iter find-index (std iter find-index), iter intersperse (std iter intersperse), iter scan (std iter scan), iter filter-map (std iter filter-map), iter flat-map (std iter flat-map), iter zip-with (std iter zip-with), log CRITICAL_LEVEL (std log CRITICAL_LEVEL), log ERROR_LEVEL (std log ERROR_LEVEL), log WARNING_LEVEL (std log WARNING_LEVEL), log INFO_LEVEL (std log INFO_LEVEL), log DEBUG_LEVEL (std log DEBUG_LEVEL), log CRITICAL_LEVEL_PREFIX (std log CRITICAL_LEVEL_PREFIX), log ERROR_LEVEL_PREFIX (std log ERROR_LEVEL_PREFIX), log WARNING_LEVEL_PREFIX (std log WARNING_LEVEL_PREFIX), log INFO_LEVEL_PREFIX (std log INFO_LEVEL_PREFIX), log DEBUG_LEVEL_PREFIX (std log DEBUG_LEVEL_PREFIX), log critical (std log critical), log error (std log error), log warning (std log warning), log info (std log info), log debug (std log debug), log custom (std log custom), assert (std assert), assert error (std assert error), assert skip (std assert skip), assert equal (std assert equal), assert not equal (std assert not equal), assert less or equal (std assert less or equal), assert less (std assert less), assert greater (std assert greater), assert greater or equal (std assert greater or equal), assert length (std assert length), assert str contains (std assert str contains), run-tests (std run-tests), xaccess (std xaccess), xupdate (std xupdate), xtype (std xtype), xinsert (std xinsert), datetime-diff (std datetime-diff), pretty-print-duration (std pretty-print-duration), path add (std path add), clip (std clip), bench (std bench), banner (std banner)

Exported aliases:
    enter, n, p, dexit, shells, g

This module exports environment.
{
    use dirs *
}
```

Also fixed up the display of exported aliases to match `help`.  You're welcome.

File: crates/nu-std/lib/help.nu [hunk 1]
```diff
@@ -123,6 +123,7 @@ def show-module [module: record] {
 
     if not ($module.commands? | is-empty) {
         print-help-header "Exported commands"
+        let indent = "    "
```
Comment (amtoine) [inline crates/nu-std/lib/help.nu:126]: this `indent` does not appear to be used in the changes :open_mouth:

File: crates/nu-std/lib/help.nu [hunk 2]
```diff
@@ -138,16 +139,16 @@ def show-module [module: record] {
     }
 
     if not ($module.aliases? | is-empty) {
-        print-help-header -n "Exported aliases:"
-        print $module.aliases
+        print-help-header "Exported aliases"
+        print $"    ($module.aliases | str join ', ')"
```
Comment (amtoine) [inline crates/nu-std/lib/help.nu:143]: > Also fixed up the display of exported aliases to match help. You're welcome.
>
out of the scope of this PR but ehe, it looks better :smirk: :sparkles: 

thanks for this simple fix :blush:

**Review by amtoine — CHANGES_REQUESTED** [review]

**Comment by fdncred** [comment]
is this ready?

**Comment by bobhy** [comment]
Yes, if @amtoine agrees.

**Review by amtoine — APPROVED** [review]
i agree :wink: 

this is a bit hacky due to the very nature of the `show-*` commands which do not return anything but rather `print` :thinking: 

i've planned on doing this for a while, so let's get rid of this very ugly error :relieved: :blush:

**Pull request merged**